olive/passes/onnx/vitis_ai/examples/torch/language_modeling/llm_ptq/quantize_quark.py

```diff
@@ -0,0 +1,461 @@
+#
+# Copyright (C) 2025, Advanced Micro Devices, Inc. All rights reserved.
+# SPDX-License-Identifier: MIT
+#
+
+# ruff: noqa: T201
+
+import argparse
+import os
+import sys
+import warnings
+from pathlib import Path
+
+import torch
+from quark.torch import ModelExporter, ModelImporter, ModelQuantizer, load_params, save_params
+from quark.torch.export.api import _move_quantizer_to_dict
+from quark.torch.utils.device import TPDeviceManager
+from transformers import AutoProcessor
+
+from olive.passes.onnx.vitis_ai.examples.torch.language_modeling.llm_ptq.configuration_preparation import (
+    get_config,
+    get_export_config,
+)
+from olive.passes.onnx.vitis_ai.examples.torch.language_modeling.llm_ptq.customized_configuration import (
+    SUPPORTED_QUANT_SCHEME,
+)
+
+# TODO: Using sys.path.append is bad practice.
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
+from olive.passes.onnx.vitis_ai.examples.torch.language_modeling.llm_eval.evaluation import eval_model
+from olive.passes.onnx.vitis_ai.examples.torch.language_modeling.llm_utils.data_preparation import get_calib_dataloader
+from olive.passes.onnx.vitis_ai.examples.torch.language_modeling.llm_utils.model_preparation import (
+    get_model,
+    get_model_type,
+    get_tokenizer,
+    prepare_for_moe_quant,
+)
+
+
+def run_quark_quantization(args: argparse.Namespace) -> None:
+    # 1. Define original model
+    print("\n[INFO]: Loading model ...")
+
+    # We currently use CPU memory to load large models because GPU memory is typically smaller.
+    # The model will be dispatched to different GPUs based on the total number of GPUs specified by torchrun --nproc-per-node.
+    # TODO:
+    # The current method results in high CPU memory consumption due to multiple copies of the same model.
+    # We plan to address this in the future by implementing a more efficient way to dispatch the model to devices.
+    if args.use_tp:
```

Comment:
```suggestion
    if args.use_tp or platform.system() == 'Windows':
```

must `import platform` beforehand

Reply:
Done. Added the platform check and import platform at the top. Windows now defaults to CPU as suggested.